server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -117,6 +117,8 @@ public class ShuffleTaskManager {
   private final ShuffleBufferManager shuffleBufferManager;
   private Map<String, ShuffleTaskInfo> shuffleTaskInfos = JavaUtils.newConcurrentMap();
   private Map<Long, PreAllocatedBufferInfo> requireBufferIds = JavaUtils.newConcurrentMap();
+  private final Map<String, Long> appsWaitingToBeRemoved = JavaUtils.newConcurrentMap();
```

Comment:
Do you really need this? 

I think we only need to remove the lock. WDYT? @xianjingfeng

Reply:
Removed. Let's fix it in another pr.